test/integration_test/fafb_matrix_test.go

```diff
@@ -0,0 +1,331 @@
+// +build integrationtest
+// +build fafb
+
+package integrationtest
+
+import (
+	"testing"
+
+	v1 "github.com/libopenstorage/operator/pkg/apis/core/v1"
+	k8sutil "github.com/libopenstorage/operator/pkg/util/k8s"
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	testutil "github.com/libopenstorage/operator/pkg/util/test"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	auto    = "auto"    // Used for things like auto-journal drives
+	size3   = "size=3"  // Recommended journal drive size, more than this is useless: https://docs.portworx.com/install-with-other/operate-and-maintain/performance-and-tuning/tuning/
+	size32  = "size=32" // Default kvdb size from spec gen
+	size50  = "size=50"
+	size100 = "size=100"
+	size200 = "size=200"
+
+	disks100 = []string{size100}
+	disks200 = []string{size200}
+
+	threeStorageNodes = uint32(3)
+
+	internalKVDB     = v1.KvdbSpec{Internal: true}
+	autoJournalDrive = &v1.CloudStorageSpec{
+		CloudStorageCommon: v1.CloudStorageCommon{
+			KvdbDeviceSpec:    &size32,
+			JournalDeviceSpec: &auto,
+			DeviceSpecs:       &disks100,
+		},
+	}
+
+	// simpleClusterSpec is a StorageCluster with internal KVDB and auto journal drives
+	simpleClusterSpec = v1.StorageClusterSpec{
+		Kvdb:         &internalKVDB,
+		CloudStorage: autoJournalDrive,
+	}
+)
+
+type TestrailCase struct {
+	ID                      string
+	Spec                    v1.StorageClusterSpec
+	BackendRequirements     BackendRequirements
+	ShouldStartSuccessfully bool
+}
+
+func (trc *TestrailCase) PopulateStorageCluster(cluster *v1.StorageCluster) {
+	cluster.Name = makeDNS1123Compatible(trc.ID)
+	cluster.Spec.CloudStorage = trc.Spec.CloudStorage
+	cluster.Spec.Kvdb = trc.Spec.Kvdb
+	cluster.Spec.Nodes = trc.Spec.Nodes
+}
+
+var flashArrayPositiveInstallTests = []TestrailCase{
+	// C55129: https://portworx.testrail.net/index.php?/cases/view/55129
+	// Install PX through operator with internal KVDB on FlashArray
+	{
+		ID:                      "C55129",
+		ShouldStartSuccessfully: true,
+		BackendRequirements: BackendRequirements{
+			RequiredArrays: 1,
+		},
+		Spec: v1.StorageClusterSpec{
```

Comment:
Let's add a TODO here:  Add the `StorageClusterSpecGen` API in the future.

Reply:
@dsupure what do you mean by that?

Reply:
nm, please ignore my comment. I was thinking to write the `StorageClusterSpecGen` that can generate a StorageCluster object instead of filling all the fields.